cli/internal/taskhash/taskhash.go

```diff
@@ -258,7 +259,14 @@ func (th *Tracker) CalculateTaskHash(pt *nodes.PackageTask, dependencySet dag.Se
 	if !ok {
 		return "", fmt.Errorf("cannot find package-file hash for %v", pkgFileHashKey)
 	}
-	hashableEnvPairs := env.GetHashableEnvPairs(pt.TaskDefinition.EnvVarDependencies)
+
+	var envPrefixes []string
+	framework := inference.InferFramework(pt.Pkg)
+	if framework != nil && framework.EnvPrefix != "" {
+		envPrefixes = append(envPrefixes, framework.EnvPrefix)
```

Comment:
Can we some verbose (debug level) logging here?

It would be helpful to be able to see _what_ is getting cached (or just a log that indicates a framework was detected and the prefix) in the case when a build becomes _uncachable_ due to an env var. 

This is also something we should probably revisit in a more holistic way in the future.

Reply:
Going to fast-follow this as our logging story isn't well-plumbed and this will need to be a more-invasive change.